.github/workflows/test-and-build.yml

```diff
@@ -6,7 +6,7 @@ on:
   # instead?
   push:
       branches:
-        - "*"
+        - "main"
```

Comment:
This workflow is already being triggered by `pull_request`. We don't need to run the workflow for all branches, because that will result in duplicate workflows for every PR.